cns/wireserver/net.go

```diff
@@ -49,6 +50,44 @@ func GetPrimaryInterfaceFromResult(res *GetInterfacesResult) (*InterfaceInfo, er
 	return nil, ErrNoPrimaryInterface
 }
 
+func GetSecondaryInterfaceFromResult(res *GetInterfacesResult) (*InterfaceInfo, error) {
+	for _, i := range res.Interface {
+		// skip if primary
+		if i.IsPrimary {
+			continue
+		}
+
+		// skip if no subnets
+		if len(i.IPSubnet) == 0 {
+			continue
+		}
+
+		// get the second subnet
+		s := i.IPSubnet[0]
+		gw, err := calculateGatewayIP(s.Prefix)
+		if err != nil {
+			return nil, err
+		}
+
+		secondaryIP := ""
```

Comment:
nit: `var string secondaryIP` instead of empty string init

Reply:
also Note: we need to pass macAddress in this function & return interface based on i.IsPrimary=false & i.MACAddress==mac